Update Red Zone to latest release (03/29/21)

Quetip opened the pull request:
Making updates to this map (work in progress).

This PR adds landing and hovering mechanics.

Closes https://github.com/OpenRA/OpenRA/issues/12596.

Quetip committed:
Update Red Zone to latest release (03/29/21)

figuring out how to make a pull request

pchote commented:
OpenRA.Utility(1,1): Error: Actor type `truk` consumes conditions that are not granted: loaded
OpenRA.Utility(1,1): Error: Actor type `fix` consumes conditions that are not granted: build-incomplete, being-demolished
OpenRA.Utility(1,1): Warning: Actor type `hind` grants conditions that are not consumed: being-captured
OpenRA.Utility(1,1): Error: rules.yaml:230 refers to a trait field `PipType` that does not exist on `Passenger`.
OpenRA.Utility(1,1): Error: rules.yaml:283 refers to a trait field `PipCount` that does not exist on `AmmoPool`.
OpenRA.Utility(1,1): Error: rules.yaml:284 refers to a trait field `PipType` that does not exist on `AmmoPool`.
OpenRA.Utility(1,1): Error: rules.yaml:314 refers to a trait field `PipCount` that does not exist on `Cargo`.
OpenRA.Utility(1,1): Error: rules.yaml:327 refers to a trait field `Scale` that does not exist on `RenderSprites`.
OpenRA.Utility(1,1): Error: rules.yaml:331 refers to a trait field `Scale` that does not exist on `RenderSprites`.
OpenRA.Utility(1,1): Error: rules.yaml:342 refers to a trait field `PipCount` that does not exist on `Cargo`.
OpenRA.Utility(1,1): Error: rules.yaml:357 refers to a trait field `Scale` that does not exist on `RenderSprites`.
OpenRA.Utility(1,1): Error: rules.yaml:377 refers to a trait field `RequiresCondition` that does not exist on `Huntable`.
OpenRA.Utility(1,1): Error: rules.yaml:385 refers to a trait field `PipCount` that does not exist on `Cargo`.
OpenRA.Utility(1,1): Error: rules.yaml:387 refers to a trait field `RequiresCondition` that does not exist on `Cargo`.
OpenRA.Utility(1,1): Error: rules.yaml:392 refers to a trait field `RequiresCondition` that does not exist on `Huntable`.
OpenRA.Utility(1,1): Error: rules.yaml:416 refers to a trait field `RequiresCondition` that does not exist on `Huntable`.
OpenRA.Utility(1,1): Error: rules.yaml:422 refers to a trait field `PipCount` that does not exist on `Cargo`.
OpenRA.Utility(1,1): Error: rules.yaml:441 refers to a trait field `RequiresCondition` that does not exist on `Huntable`.
OpenRA.Utility(1,1): Error: rules.yaml:447 refers to a trait field `PipCount` that does not exist on `Cargo`.
OpenRA.Utility(1,1): Error: rules.yaml:460 refers to a trait field `RequiresCondition` that does not exist on `Huntable`.
OpenRA.Utility(1,1): Error: rules.yaml:466 refers to a trait field `PipCount` that does not exist on `Cargo`.
OpenRA.Utility(1,1): Error: weapons.yaml:34 refers to a weapon field `BurstDelay` that does not exist.
OpenRA.Utility(1,1): Error: weapons.yaml:40 refers to a projectile field `Angle` that does not exist on `Bullet`.
OpenRA.Utility(1,1): Error: weapons.yaml:120 refers to a warhead field `ImpactSound` that does not exist on `CreateEffect`.

Quetip commented:
acknowledged.

Quetip committed:
Enter and Exit Hind logic

Implemented new enter and exit logic for Hind gunship. Preciously code would not allow aircraft the player to switch ownership causing an exited player to still "own" the gunship. Additional help from hamb.

Quetip commented:
Can I reopen this and make changes to or just open a new PR?

abcdefg30 commented:
Whatever suits you better. (You can do both.)

Smittytron commented on mods/ra/maps/red-zone/map.yaml:
```diff
@@ -0,0 +1,3352 @@
+		Color: ABB7E4
+		Allies: England Northern Base, England Air Base, England Prison Base, England Missile Silos
+		Enemies: Creeps, USSR, BadGuy
+	PlayerReference@Creeps:
```
Creeps can be removed if they aren't in this mission.

Smittytron commented on mods/ra/maps/red-zone/map.yaml:
```diff
@@ -0,0 +1,3352 @@
+		Enemies: Creeps, England Missile Silos, England Prison Base, England Main Base, England Northern Base, England Air Base
+
+Actors:
+	Actor1243: cycl
```
This is a pain, but all fences and walls need to be assigned a correct owner for targeting purposes. (England I assume)

Smittytron commented on mods/ra/maps/red-zone/red-zone-AI.lua:
```diff
@@ -0,0 +1,336 @@
+﻿--[[
+   Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
```
2021

Smittytron commented on mods/ra/maps/red-zone/red-zone.lua:
```diff
@@ -0,0 +1,608 @@
+﻿--[[
+   Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
```
2021

Quetip replied:
Updated to 2022

Smittytron commented on mods/ra/maps/red-zone/red-zone.lua:
```diff
@@ -0,0 +1,608 @@
+
+EnglandAirBaseReveal = function()
+	if not EnglandAirBaseRevealed then
+		Media.DisplayMessage("Hijack a Hind Helicopter.")
```
Considering we changed these to Blackhawks for Skirmish/Multiplayer I'd go ahead a switch this to the UH60.

Smittytron commented on mods/ra/maps/red-zone/red-zone.lua:
```diff
@@ -0,0 +1,608 @@
+		return PlayerIsHumanOrBot(p)
+	end)
+
+	ussr = Player.GetPlayer("USSR")
```
Non `local` references should be capitalized.

Smittytron commented on mods/ra/maps/red-zone/red-zone.lua:
```diff
@@ -0,0 +1,608 @@
+		end)
+	end)
+
+	AlliedObjective = england_main.AddPrimaryObjective("Don't let soviets to destroy the Missile Silos.", england_main.Color)
```
Use `AddObjective` instead. We are phasing out `AddPrimaryObjective` and `AddSecondaryObjective`.

Smittytron commented on mods/ra/maps/red-zone/rules.yaml:
```diff
@@ -0,0 +1,490 @@
+		Weapon: Atomic
+	-TooltipDescription@ally:
+	-TooltipDescription@other:
+
```
Extra space.

Quetip replied:
We can discuss/implement these changes. I'm working on the helicopter logic now.

Smittytron reviewed (commented):
In general I want us to cut down on the custom rules and weapons as much as possible. If the mission doesn't need it to function I'd like to cut it out.

The map.yaml is a hefty 3300+ lines, which I guess is fine but if you see any actors that look superfluous I'd go ahead and make cuts there too.

Quetip commented:
Right now I'm getting it to work. At the moment, I working to get the Heli to idle in the air with out it auto landing. Setting "IdleBehavior" trait to **None** will spawn the helicopter hovering in air and the user is unable to enter the unit. So, changes need to be made somehow. There's a few mechanics that have changed with the helicopter since the initial release of this map.

Quetip committed:
Changes in Helicopter Logic

Made changes in Hind logic in terms of entering and exiting the vehicle(WIP). Deleted creeps. Changed ownership of walls and fences to England. Minor cosmetic changes.

Quetip committed:
Global Variables Capitalized

Made updates:
 Capitalized globals.
 Removed unused files.
 Updated ownership for walls
 Updated dates.

Quetip commented:
FWIW not sure If I can pick up where this left off BUT I made these updates. I could not get the Helicopter logic to work.

Quetip committed:
Local Machine Copy

Changing a fork to mimic local machine. latest version.

PunkPun commented:
Closing due to inactivity. 

It also appears like this PR was corrupted, perhaps by the last commit?
